pinot-core/src/main/java/org/apache/pinot/core/data/manager/offline/DimensionTableDataManager.java

```diff
@@ -75,16 +72,10 @@ public static DimensionTableDataManager getInstanceByTableName(String tableNameW
     return INSTANCES.get(tableNameWithType);
   }
 
-  /**
-   * Instance properties/methods
-   */
-  private final ReadWriteLock _rwl = new ReentrantReadWriteLock();
-  private final Lock _lookupTableReadLock = _rwl.readLock();
-  private final Lock _lookupTableWriteLock = _rwl.writeLock();
-
-  // _lookupTable is a HashMap used for storing/serving records for a table keyed by table PK
-  @GuardedBy("_rwl")
-  private final Map<PrimaryKey, GenericRow> _lookupTable = new HashMap<>();
+  @SuppressWarnings("rawtypes")
+  private static final AtomicReferenceFieldUpdater<DimensionTableDataManager, Map> UPDATER =
```

Comment:
Since we have `_lookupTable` as volatile, do we still need this updater

Reply:
Yes, for the CAS